pkg/registry/core/pod/storage/storage.go

```diff
@@ -229,3 +235,96 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 	// subresources should never allow create on update.
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation, false, options)
 }
+
+// EphemeralContainersREST implements the REST endpoint for adding EphemeralContainers
+type EphemeralContainersREST struct {
+	store *genericregistry.Store
+}
+
+var _ = rest.Patcher(&EphemeralContainersREST{})
+
+// Get of this endpoint will return the list of ephemeral containers in this pod
+func (r *EphemeralContainersREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
+		return nil, errors.NewBadRequest("feature EphemeralContainers disabled")
+	}
+
+	obj, err := r.store.Get(ctx, name, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return ephemeralContainersInPod(obj.(*api.Pod)), nil
+}
+
+// New creates a new EphemeralContainers resource
+func (r *EphemeralContainersREST) New() runtime.Object {
+	return &api.EphemeralContainers{}
+}
+
+// Update alters the EphemeralContainers field in PodSpec
+func (r *EphemeralContainersREST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.EphemeralContainers) {
+		return nil, false, errors.NewBadRequest("feature EphemeralContainers disabled")
+	}
+
+	obj, err := r.store.Get(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+	pod := obj.(*api.Pod)
+
+	// Build an UpdatedObjectInfo to pass to the pod store.
+	// It is given the currently stored v1.Pod and transforms it to the new pod that should be stored.
+	updatedPodInfo := rest.DefaultUpdatedObjectInfo(pod, func(ctx context.Context, oldObject, _ runtime.Object) (newObject runtime.Object, err error) {
```

Comment:
This probably deserves an integration test, can be a follow up (do you have a follow up list in the parent issue)?

Reply:
np, I've already added some integration tests in PR #79216. 

I've added a list of follow up PRs to the description of this PR, and a list of all PRs in the parent issue.